source/MaterialXCore/Interface.cpp

```diff
@@ -301,6 +304,23 @@ NodePtr Input::getConnectedNode() const
     return PortElement::getConnectedNode();
 }
 
+void Input::setInterfaceInput(const string& interfaceName)
+{
+    if (interfaceName != EMPTY_STRING)
```

Comment:
Does proper validity checks for existence of interface and clears properly by removing the `interfacename` (disallows setting to empty string).